pubsub/subscriptions/pull_settings.go

```diff
@@ -34,7 +34,18 @@ func pullMsgsSettings(w io.Writer, projectID, subName string) error {
 
 	sub := client.Subscription(subName)
 	sub.ReceiveSettings.Synchronous = true
+	// MaxOutstandingMessages is the maximum number of unprocessed messages the
+	// client will pull from the server before pausing.
+	//
+	// This is only guaranteed when ReceiveSettings.Synchronous is set to true.
+	// When Synchronous is set to false, the StreamingPull RPC is used which
+	// can pull a single large batch of messages at once that is greater than
+	// MaxOustandingMessages before pausing. For more info, see
+	// https://cloud.google.com/pubsub/docs/pull#streamingpull_dealing_with_large_backlogs_of_small_messages.
 	sub.ReceiveSettings.MaxOutstandingMessages = 10
+	// MaxOutstandingBytes is the maximum size of unprocessed messages,
+	// that the client will pull from the server before pausing.
```

Comment:
Does this also have the caveat above with `Synchronous`? Maybe say to `See MaxOutstandingMessages`?